### ISSUE TITLE
Failed to handle method call: java.lang.NullPointerException

### ISSUE THREAD

**Issue by quetool** [RodrigoSMarques/flutter_branch_sdk#188]
Hello! I didn't want to open a new issue but I don't receive any answers on other similar issues I commented so here we go.

I get this error when calling `FlutterBranchSdk.showShareSheet()`
I recently updated to latest Flutter but it was happening the same on version 3.0.1.

```
[+2468 ms] E/MethodChannel#flutter_branch_sdk/message(20572): Failed to handle method call
[        ] E/MethodChannel#flutter_branch_sdk/message(20572): java.lang.NullPointerException: Attempt to invoke virtual method 'java.lang.String java.lang.Object.toString()' on a null object reference
[        ] E/MethodChannel#flutter_branch_sdk/message(20572): 	at br.com.rsmarques.flutter_branch_sdk.FlutterBranchSdkHelper.convertToBUO(FlutterBranchSdkHelper.java:133)
[        ] E/MethodChannel#flutter_branch_sdk/message(20572): 	at br.com.rsmarques.flutter_branch_sdk.FlutterBranchSdkPlugin.showShareSheet(FlutterBranchSdkPlugin.java:381)
[        ] E/MethodChannel#flutter_branch_sdk/message(20572): 	at br.com.rsmarques.flutter_branch_sdk.FlutterBranchSdkPlugin.onMethodCall(FlutterBranchSdkPlugin.java:241)
[        ] E/MethodChannel#flutter_branch_sdk/message(20572): 	at io.flutter.plugin.common.MethodChannel$IncomingMethodCallHandler.onMessage(MethodChannel.java:262)
[        ] E/MethodChannel#flutter_branch_sdk/message(20572): 	at io.flutter.embedding.engine.dart.DartMessenger.invokeHandler(DartMessenger.java:295)
[        ] E/MethodChannel#flutter_branch_sdk/message(20572): 	at io.flutter.embedding.engine.dart.DartMessenger.lambda$dispatchMessageToQueue$0$DartMessenger(DartMessenger.java:319)
[        ] E/MethodChannel#flutter_branch_sdk/message(20572): 	at io.flutter.embedding.engine.dart.-$$Lambda$DartMessenger$AIEPqY6mWzaNK15HekX9bftoAXs.run(Unknown Source:12)
[        ] E/MethodChannel#flutter_branch_sdk/message(20572): 	at android.os.Handler.handleCallback(Handler.java:938)
[        ] E/MethodChannel#flutter_branch_sdk/message(20572): 	at android.os.Handler.dispatchMessage(Handler.java:99)
[        ] E/MethodChannel#flutter_branch_sdk/message(20572): 	at android.os.Looper.loop(Looper.java:255)
[        ] E/MethodChannel#flutter_branch_sdk/message(20572): 	at android.app.ActivityThread.main(ActivityThread.java:8224)
[        ] E/MethodChannel#flutter_branch_sdk/message(20572): 	at java.lang.reflect.Method.invoke(Native Method)
[        ] E/MethodChannel#flutter_branch_sdk/message(20572): 	at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:632)
[        ] E/MethodChannel#flutter_branch_sdk/message(20572): 	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:1049)
[  +45 ms] I/flutter (20572): --- Sentry Error in Development ---
[   +1 ms] I/flutter (20572):
[        ] I/flutter (20572): PlatformException(error, Attempt to invoke virtual method 'java.lang.String java.lang.Object.toString()' on a null object reference, null, java.lang.NullPointerException: Attempt to invoke virtual method 'java.lang.String java.lang.Object.toString()' on a null object reference
[        ] I/flutter (20572): 	at br.com.rsmarques.flutter_branch_sdk.FlutterBranchSdkHelper.convertToBUO(FlutterBranchSdkHelper.java:133)
[        ] I/flutter (20572): 	at br.com.rsmarques.flutter_branch_sdk.FlutterBranchSdkPlugin.showShareSheet(FlutterBranchSdkPlugin.java:381)
[        ] I/flutter (20572): 	at br.com.rsmarques.flutter_branch_sdk.FlutterBranchSdkPlugin.onMethodCall(FlutterBranchSdkPlugin.java:241)
[        ] I/flutter (20572): 	at io.flutter.plugin.common.MethodChannel$IncomingMethodCallHandler.onMessage(MethodChannel.java:262)
[   +5 ms] I/flutter (20572): 	at io.flutter.embedding.engine.dart.DartMessenger.invokeHandler(DartMessenger.java:295)
[        ] I/flutter (20572): 	at io.flutter.embedding.engine.dart.DartMessenger.lambda$dispatchMessageToQueue$0$DartMessenger(DartMessenger.java:319)
[        ] I/flutter (20572): 	at io.flutter.embedding.engine.dart.-$$Lambda$DartMessenger$AIEPqY6mWzaNK15HekX9bftoAXs.run(Unknown Source:12)
[        ] I/flutter (20572): 	at android.os.Handler.handleCallback(Handler.java:938)
[        ] I/flutter (20572): 	at android.os.Handler.dispatchMessage(Handler.java:99)
[        ] I/flutter (20572): 	at android.os.Looper.loop(Looper.java:255)
[        ] I/flutter (20572): 	at android.app.ActivityThread.main(ActivityThread.java:8224)
[        ] I/flutter (20572): 	at java.lang.reflect.Method.invoke(Native Method)
[        ] I/flutter (20572): 	at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:632)
[        ] I/flutter (20572): 	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:1049)
[        ] I/flutter (20572): )
```

Flutter doctor -v

```
[✓] Flutter (Channel stable, 3.3.7, on macOS 12.5.1 21G83 darwin-arm, locale en-ES)
    • Flutter version 3.3.7 on channel stable at /Users/alfreedom/Development/flutter
    • Upstream repository https://github.com/flutter/flutter.git
    • Framework revision e99c9c7cd9 (6 days ago), 2022-11-01 16:59:00 -0700
    • Engine revision 857bd6b74c
    • Dart version 2.18.4
    • DevTools version 2.15.0

[✓] Android toolchain - develop for Android devices (Android SDK version 32.1.0-rc1)
    • Android SDK at /Users/alfreedom/Library/Android/sdk
    • Platform android-33, build-tools 32.1.0-rc1
    • Java binary at: /Applications/Android Studio.app/Contents/jre/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build 11.0.11+0-b60-7772763)
    • All Android licenses accepted.

[✓] Xcode - develop for iOS and macOS (Xcode 14.1)
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • Build 14B47b
    • CocoaPods version 1.11.3

[✓] Chrome - develop for the web
    • Chrome at /Applications/Google Chrome.app/Contents/MacOS/Google Chrome

[✓] Android Studio (version 2021.1)
    • Android Studio at /Applications/Android Studio.app/Contents
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build 11.0.11+0-b60-7772763)

[✓] VS Code (version 1.73.0)
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Flutter extension version 3.52.0

[✓] HTTP Host Availability
    • All required HTTP hosts are available

• No issues found!
```

**Comment by RodrigoSMarques** [comment]
Hi @quetool 

You are sending `null` in some value of a `CustomMetadata` (`addCustomMetadata` method).

I will insert a consistency in the next version.

**Comment by quetool** [comment]
Hey @RodrigoSMarques, that's not happening, no way I'm sending null values here:

This is the BUO I create

```
return BranchUniversalObject(
  canonicalIdentifier: linkPath,
  canonicalUrl: canonicalUrl,
  title: linkTitle,
  imageUrl: linkImage,
  contentDescription: linkDescription,
  contentMetadata: BranchContentMetaData()
    ..addCustomMetadata('username', user?.formattedNickname ?? '')
    ..addCustomMetadata('post_id', post?.id ?? '')
    ..addCustomMetadata('path', linkPath),
);
```

No way those 3 customs metadata are null

**Comment by RodrigoSMarques** [comment]
The stacktrace indicates that it occurred when sending the CustomMetadata to the Native SDK, due to a null value.

What is the value of `linkPath`? it does not have the ternary operator.

**Comment by RodrigoSMarques** [comment]
Fix completed - Version 6.3.0.
